x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ShrinkAction.java

```diff
@@ -57,16 +59,22 @@ public class ShrinkAction implements LifecycleAction {
             MAX_PRIMARY_SHARD_SIZE,
             ObjectParser.ValueType.STRING
         );
+        PARSER.declareBoolean(ConstructingObjectParser.optionalConstructorArg(), ALLOW_WRITES_ON_TARGET);
     }
 
     private Integer numberOfShards;
     private ByteSizeValue maxPrimaryShardSize;
+    private Boolean allowWritesOnTarget;
 
     public static ShrinkAction parse(XContentParser parser) throws IOException {
         return PARSER.parse(parser, null);
     }
 
-    public ShrinkAction(@Nullable Integer numberOfShards, @Nullable ByteSizeValue maxPrimaryShardSize) {
+    public ShrinkAction(
+        @Nullable Integer numberOfShards,
+        @Nullable ByteSizeValue maxPrimaryShardSize,
+        @Nullable Boolean allowWritesOnTarget
```

Comment:
My preference here would be to keep this as a boolean, when we are reading from the wire and its absent we fallback to the previous behaviour. I think the default behaviour is clear enough and this way we can avoid some null checks in the code.

What do you think?

Reply:
I think that makes sense, but there's a scenario where it might be useful to have null. If a policy is created with the `allowWritesOnTarget` absent from the json, and we set the default value of false in the action. Then if the user calls `GET /_ilm/policy/my-policy` , we have to return the policy with the default value of false, rather than with the field absent. This might be confusing for the user. On the other hand, it might also be instructive to see the defaults included when the policy is accessed :thinking: . What do you think?

Reply:
This is a very good point. Now for this I do not have a good answer apart so I will delegate to "the wisdom of our ancestors", meaning I will check what the other actions do.

So, if my analysis is correct, please double check, optional objects are indeed nullable but optional booleans are not. This are the actions I stumbled across.
- [Delete](https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-delete.html)
- [Migrate](https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-migrate.html)
- [Searchable snapshot](https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-searchable-snapshot.html)

Please check as well, and if it did they all behave like this, I think it would make sense to keep it.

Reply:
Ahh, good thinking to look into those :+1:  You're right, those have regular non-nullable booleans, and the defaults are returned by `GET`. So that answer that question, thanks!